src/components/lib/posix/posix.c

```diff
@@ -31,7 +34,7 @@ libc_syscall_override(cos_syscall_t fn, int syscall_num)
 	cos_syscalls[syscall_num] = fn;
 }
 
-struct sl_lock stdout_lock = SL_LOCK_STATIC_INIT();
+//struct sl_lock stdout_lock = SL_LOCK_STATIC_INIT();
```

Comment:
Is there demand for a single threaded version of this? Removing all this synchronization logic is going to lead to weird bugs as soon as you have more than one thread. Why doesn't it work as-is? Too tight coupling to SL?

Reply:
Great question, and good intuition. We revamping libposix right now to use a generic interface that can either be co-located (via library) with sl, *or* accessed in another component. Given this, most of the synchronization facilities (and sl dependencies) in posix need to change backing APIs.

As such, Bite is trying to get sqlite working for a comparison case, and is not trying to solve other problems for now. As we update posix, I'm hoping we can add back in those features.

A lot of the cruft assuming single-component runtimes has been removed, but we have more to do, and libposix is one of the big places more love needs to be applied.